### DISCUSSION TITLE
Auth0 not working

### DISCUSSION THREAD

**Discussion by mforell** [langfuse/langfuse#2165] [Support]
Hi ,
Hey everyone,

I'm running a self-hosted instance of Langfuse in Docker, and I'm having trouble integrating Auth0. I followed the setup instructions, but when I try to authenticate, I get this error in my logs:

```
Unknown argument `refresh_expires_in`. Available options are marked with ?. PrismaClientValidationError:

langfuse-server-1  | Invalid `prisma.account.create()` invocation:
langfuse-server-1  | Unknown argument `refresh_expires_in`. Available options are marked with ?.
langfuse-server-1  |     at Tn (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:115:6855)
langfuse-server-1  |     at In.handleRequestError (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:6510)
langfuse-server-1  |     at In.handleAndLogRequestError (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:6188)
langfuse-server-1  |     at In.request (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:5896)
langfuse-server-1  |     at async l (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:127:11167) {
langfuse-server-1  |   name: 'LinkAccountError',
langfuse-server-1  |   code: undefined
langfuse-server-1  | }

```
Has anyone else encountered this issue? Any tips on how to resolve it would be greatly appreciated!

Thanks!

**Comment by marcklingen** [comment]
Does a user with the same email already exist in the db?

**Reply by mforell** [reply]
no. I also started with a new container and volume in docker.

**Reply by marcklingen** [reply]
that's interesting. There are a couple of teams who I know that run on Auth0, can you grant additional permissions in Auth0 to try if this solves the problem?

**Reply by mforell** [reply]
I am not sure how to grant more permossions with docker-compose for Auth0

**Reply by marcklingen** [reply]
on the Auth0 side, are there any configuration options?

I am just guessing what could be the issue here, as I verified that Auth0 works when initially adding support for it and as I know some teams who use Auth0 SSO with Langfuse. If you find the solution to this issue, I'd love to add it to our FAQs: https://langfuse.com/faq

**Reply by mforell** [reply]
ah ok. Thank you very much. I'll have a look 

yes, of course :)

**Reply by marcklingen** [reply]
thank you!

**Comment by damianradowiecki** [comment]
A similar issue here, when trying to use Keycloak as SSO:

```
Unknown argument `refresh_expires_in`. Available options are marked with ?.
2024-06-26T13:32:17.489408698Z     at Tn (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:115:6855)
2024-06-26T13:32:17.489448031Z     at In.handleRequestError (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:6510)
2024-06-26T13:32:17.489464233Z     at In.handleAndLogRequestError (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:6188)
2024-06-26T13:32:17.489500137Z     at In.request (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:5896)
2024-06-26T13:32:17.489553866Z     at async l (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:127:11167) {
2024-06-26T13:32:17.489565105Z   name: 'LinkAccountError',
2024-06-26T13:32:17.489589556Z   code: undefined
2024-06-26T13:32:17.489657037Z }
```

**Reply by marcklingen** [reply]
Keycloak is not supported as of now ([docs](https://langfuse.com/docs/deployment/self-host#sso)), how did you configure Langfuse?

There is a closed [PR](https://github.com/langfuse/langfuse/pull/2224), are you working on this?

**Reply by damianradowiecki** [reply]
I used custom OAuth variables (listed in the link you've sent):
```
AUTH_CUSTOM_CLIENT_ID
AUTH_CUSTOM_CLIENT_SECRET
AUTH_CUSTOM_ISSUER
AUTH_CUSTOM_NAME
```

However, it does not work.

I haven't noticed this PR actually. The PR looks like almost ready. Do you know why it has been closed?

**Reply by marcklingen** [reply]
I'd assume that you can fix this by changing the scopes of the custom oauth setup enabled by this change, to be released today: https://github.com/langfuse/langfuse/pull/2459

Otherwise, can you contribute a Keycloak setup? I do not know why the original PR was closed

**Reply by damianradowiecki** [reply]
I will check if it helps.

Regarding contribution - if the provided solution does not work, I will take a look at the code.

**Reply by marcklingen** [reply]
did you get to try the custom OAuth route yet?

**Reply by damianradowiecki** [reply]
Yes. I've tried the latest langfuse image. I still get the same error.

**Reply by marcklingen** [reply]
ok, then let's go for official support of keycloak, open for PR on this

**Reply by RTae** [reply]
@damianradowiecki Can you solve this issue ? I facing same issue with you. I set Keycloak as SSO

**Reply by beratcberkol** [reply]
Is there any update on Keycloak SSO?

**Reply by marcklingen** [reply]
Still open for PR to add it to langfuse

**Reply by damianradowiecki** [reply]
> @damianradowiecki Can you solve this issue ? I facing same issue with you. I set Keycloak as SSO

Hey @RTae, not in the near future. If I find more time I will open a PR

**Reply by RTae** [reply]
@damianradowiecki Thank you. That's perfect. Currently, I'm working on it. I think we could share a PR

**Reply by RTae** [reply]
@damianradowiecki  @bbartels  @marcklingen  Could you help me review this PR. I just made an update to support Keycloak.

https://github.com/langfuse/langfuse/pull/2866

**Reply by marcklingen** [reply]
thanks, added comments. also there is this additional PR: #2846

**Reply by marcklingen** [reply]
Native Keycloak support was released today: https://langfuse.com/changelog/2024-12-03-new-sso-providers-github-enterprise-and-keycloak

**Comment by maithilee25** [comment]
Hi,
I am facing a similar issue with a self-hosted instance of Langfuse with SSO integration with OAuth2 identity server.
```
Unknown argument `jti`. Did you mean `id`? Available options are marked with ?.
    at Tn (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:115:6855)
    at In.handleRequestError (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:6510)
    at In.handleAndLogRequestError (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:6188)
    at In.request (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:122:5896)
    at async l (/app/node_modules/.pnpm/@prisma+client@5.13.0_prisma@5.13.0/node_modules/@prisma/client/runtime/library.js:127:11167) {
  name: 'LinkAccountError',
  code: undefined

```

**Reply by RTae** [reply]
Similarly, this is quite similar to the KeyCloak case. You need to find a way to not return JTI data from your authentication provider.

**Reply by maithilee25** [reply]
It won't be possible to update the response by the IdP.
Probably need to implement something like [this](https://github.com/langfuse/langfuse/pull/2846/files#diff-e58e5990c970014f04ffaa1ab621f49c03a311caff05ded1408ecfb18f5208ebR266) to remove the additional fields.

Is there any plan for a generic solution since it looks like integration with other IdPs too are facing this issue?

**Reply by RTae** [reply]
Yeah, currently I'm working on Keycloak. I removed unnecessary variable from the payload before save data into a database ( https://github.com/langfuse/langfuse/pull/2866 ). I think if you would like to contribute, Langfuse will support your current system IdP,  you can open a PR.

**Reply by marcklingen** [reply]
Thanks again for your contribution (and sorry for moving slow on the review, accidentally muted the email thread)!

Released today: https://langfuse.com/changelog/2024-12-03-new-sso-providers-github-enterprise-and-keycloak
